src/Illuminate/Auth/DatabaseUserProvider.php

```diff
@@ -70,7 +70,9 @@ public function retrieveByToken($identifier, $token)
     {
         $user = $this->conn->table($this->table)->find($identifier);
 
-        return $user && hash_equals($user->remember_token, $token)
+        $rememberToken = $user->remember_token;
```

Comment:
@mcordingley @taylorotwell 
Same question as in the [other PR](https://github.com/laravel/framework/pull/21328#pullrequestreview-64722044): What is if `$user == null`, i.e. the user got deleted from our database? Or is that not possible? But then we wouldn't need the null-check below, too.

Reply:
That sounds entirely possible. I am nowhere near a computer right now. Add a test for it to both places so we'll know if it's an issue now and in the future and rework the code so it passes if it really is a problem.

Reply:
I think in the normal course of operations, we wouldn't be attempting a token-based login for a user who has gone away. But, a deleted user could trigger this and it looks like it would error. 

I can submit another PR tomorrow at the latest to add the tests and address this, if you don't beat me to the punch.

Reply:
Addressed in #21368.